### PR TITLE
Fix seqno->time worker not scheduled with multi DB instances

### PR DESCRIPTION
**Pull request by jay-zhuang**
`PeriodicWorkScheduler` is a global singleton, which were used to store per-instance setting `record_seqno_time_cadence_`. Move that to db_impl.h which is per-instance.

### CODE REVIEW

File: db/periodic_work_scheduler.cc [hunk 1]
```diff
@@ -64,14 +64,6 @@ Status PeriodicWorkScheduler::Register(DBImpl* dbi,
 Status PeriodicWorkScheduler::RegisterRecordSeqnoTimeWorker(
     DBImpl* dbi, uint64_t record_cadence_sec) {
   MutexLock l(&timer_mu_);
-  if (record_seqno_time_cadence_ == record_cadence_sec) {
```
Comment (siying) [inline db/periodic_work_scheduler.cc:67]: Do we still do this check after the change?

File: db/db_impl/db_impl.cc [hunk 2]
```diff
@@ -823,8 +823,16 @@ Status DBImpl::RegisterRecordSeqnoTimeWorker() {
...
-      this, seqno_time_cadence);
+  Status s;
+  if (seqno_time_cadence == 0) {
+    periodic_work_scheduler_->UnregisterRecordSeqnoTimeWorker(this);
+  } else {
+    s = periodic_work_scheduler_->RegisterRecordSeqnoTimeWorker(
+        this, seqno_time_cadence);
+  }
+  if (s.ok()) {
+    record_seqno_time_cadence_ = seqno_time_cadence;
```
Comment (siying) [inline db/db_impl/db_impl.cc:834]: I see the variable set, but didn't see how it used.

File: db/periodic_work_scheduler.cc (hunk 1 above)
Comment (jay-zhuang) [reply db/periodic_work_scheduler.cc:67]: Missed that, added.

File: db/db_impl/db_impl.cc (hunk 2 above)
Comment (jay-zhuang) [reply db/db_impl/db_impl.cc:834]: Yeah, it should be used to compare. Added, the same as above.

**Pull request closed without merging**